repos/os/include/i2c_session/client.h

```diff
@@ -0,0 +1,58 @@
+/*
+ * \brief  I2C session client implementation
+ * \author Jean-Adrien Domage <jean-adrien.domage@gapfruit.com>
+ * \date   2021-02-25
+ */
+
+/*                                                                              
+ * Copyright (C) 2013-2021 Genode Labs GmbH                                     
+ * Copyright (C) 2021 gapfruit AG
+ *
+ * This file is part of the Genode OS framework, which is distributed
+ * under the terms of the GNU Affero General Public License version 3.
+ */
+
+#ifndef _INCLUDE__I2C_SESSION__CLIENT_H_
+#define _INCLUDE__I2C_SESSION__CLIENT_H_
+
+#include <base/rpc_client.h>
+#include <i2c_session/capability.h>
+
+namespace I2c {
+
+	class Session_client;
+
+}
+
+class I2c::Session_client : public Genode::Rpc_client<I2c::Session>
+{
+public:
+
+	explicit Session_client(I2c::Session_capability session)
+	:
+		Genode::Rpc_client<I2c::Session>(session)
+	{}
+
+	void write_byte(Genode::uint8_t byte) override
+	{
+		call<Rpc_write_byte>(byte);
+	}
+
+	Genode::uint8_t read_byte() override
+	{
+		return call<Rpc_read_byte>();
+	}
+
+	void write_word(Genode::uint16_t word) override
```

Comment:
I would not name it `write_word` and `read_word`, because this naming scheme is ancient and misleading, because it is dependent on the processor architecture. I think `read_16bit` and `read_8bit`, as well as `write_16bit` and `write_8bit` are much better. It is good to read and unified.